src/cascadia/TerminalApp/GlobalAppSettings.h

```diff
@@ -27,73 +28,74 @@ namespace TerminalAppLocalTests
-    void LayerJson(const Json::Value& json);
-
-    void ApplyToSettings(winrt::TerminalApp::TerminalSettings& settings) const noexcept;
-
-    std::vector<TerminalApp::SettingsLoadWarnings> GetKeybindingsWarnings() const;
-
-    const winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::TerminalApp::Command>& GetCommands() const noexcept;
-    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::TerminalApp::Command>& GetCommands() noexcept;
-
-    // These are implemented manually to handle the string/GUID exchange
-    // by higher layers in the app.
-    void DefaultProfile(const GUID defaultProfile) noexcept;
-    GUID DefaultProfile() const;
-    std::optional<std::wstring> UnparsedDefaultProfile() const;
-
-    GETSET_PROPERTY(int32_t, InitialRows); // default value set in constructor
-    GETSET_PROPERTY(int32_t, InitialCols); // default value set in constructor
-    GETSET_PROPERTY(bool, AlwaysShowTabs, true);
-    GETSET_PROPERTY(bool, ShowTitleInTitlebar, true);
-    GETSET_PROPERTY(bool, ConfirmCloseAllTabs, true);
-    GETSET_PROPERTY(winrt::Windows::UI::Xaml::ElementTheme, Theme, winrt::Windows::UI::Xaml::ElementTheme::Default);
-    GETSET_PROPERTY(winrt::Microsoft::UI::Xaml::Controls::TabViewWidthMode, TabWidthMode, winrt::Microsoft::UI::Xaml::Controls::TabViewWidthMode::Equal);
-    GETSET_PROPERTY(bool, ShowTabsInTitlebar, true);
-    GETSET_PROPERTY(std::wstring, WordDelimiters); // default value set in constructor
-    GETSET_PROPERTY(bool, CopyOnSelect, false);
-    GETSET_PROPERTY(winrt::Microsoft::Terminal::TerminalControl::CopyFormat, CopyFormatting, 0);
-    GETSET_PROPERTY(bool, WarnAboutLargePaste, true);
-    GETSET_PROPERTY(bool, WarnAboutMultiLinePaste, true);
-    GETSET_PROPERTY(LaunchPosition, InitialPosition);
-    GETSET_PROPERTY(winrt::TerminalApp::LaunchMode, LaunchMode, winrt::TerminalApp::LaunchMode::DefaultMode);
-    GETSET_PROPERTY(bool, SnapToGridOnResize, true);
-    GETSET_PROPERTY(bool, ForceFullRepaintRendering, false);
-    GETSET_PROPERTY(bool, SoftwareRendering, false);
-    GETSET_PROPERTY(bool, ForceVTInput, false);
-    GETSET_PROPERTY(bool, DebugFeaturesEnabled); // default value set in constructor
-    GETSET_PROPERTY(bool, StartOnUserLogin, false);
-    GETSET_PROPERTY(bool, AlwaysOnTop, false);
-    GETSET_PROPERTY(bool, UseTabSwitcher, true);
-
-private:
-    std::optional<std::wstring> _unparsedDefaultProfile;
-    GUID _defaultProfile;
-
-    winrt::com_ptr<winrt::TerminalApp::implementation::AppKeyBindings> _keybindings;
-    std::vector<::TerminalApp::SettingsLoadWarnings> _keybindingsWarnings;
-
-    std::unordered_map<std::wstring, winrt::TerminalApp::ColorScheme> _colorSchemes;
-    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::TerminalApp::Command> _commands;
-
-    friend class TerminalAppLocalTests::SettingsTests;
-    friend class TerminalAppLocalTests::ColorSchemeTests;
-};
+    BASIC_FACTORY(GlobalAppSettings);
+}

```

Comment:
no need for factory -- you will never construct one of these by calling the ctor!